tests/kerastuner/engine/tuner_workflows_test.py

```diff
@@ -152,7 +153,8 @@ def test_hypermodel_with_dynamic_space(tmp_dir):
         executions_per_trial=3,
         directory=tmp_dir)
 
-    assert tuner.hypermodel == hypermodel
+    # tuner.hypermodel is a KerasModelWrapper
```

Comment:
This is likely to be confusing for anyone accessing `tuner.hypermodel`

Reply:
We already do some wrapping (turning callables into HyperModel instances), I think while it could be confusing it is less confusing than having `tuner.hypermodel.build` not work properly (especially given the concerns in the first comment above)